comma/hypothesis.py

```diff
@@ -26,63 +30,312 @@ class Hypothesis:
-        'children_presence_no',
-        'children_presence_yes',
-        'housing_financial_difficulties_no',
-        'housing_financial_difficulties_yes',
-        'selfrated_health_average',
-        'selfrated_health_good',
-        'selfrated_health_poor',
-        'critical_job_no',
-        'critical_job_yes'
+        "age_group__1",
+        "age_group__2",
+        "age_group__3",
+        "age_group__4",
+        "gender_f",
+        "gender_m",
+        "education_high",
+        "education_low",
+        "education_medium",
+        "unemployed_no",
+        "unemployed_yes",
+        "have_partner_no",
+        "have_partner_yes",
+        "depressed_no",
+        "depressed_yes",
+        "children_presence_no",
+        "children_presence_yes",
+        "housing_financial_difficulties_no",
+        "housing_financial_difficulties_yes",
+        "selfrated_health_average",
+        "selfrated_health_good",
+        "selfrated_health_poor",
+        "critical_job_no",
+        "critical_job_yes",
     ]
 
     all_possible_actions = [
-        'work_from_home',
-        'maintain_physical_distance',
-        'stay_at_home',
-        'exercise',
-        'socialise',
-        'travel',
-        'seek_help',
-        'negative_coping',
-        'positive_coping',
-        'socialise_online'
+        "work_from_home",
+        "maintain_physical_distance",
+        "stay_at_home",
+        "exercise",
+        "socialise",
+        "travel",
+        "seek_help",
+        "negative_coping",
+        "positive_coping",
+        "socialise_online",
     ]
 
+    @staticmethod
+    def get_file_paths(url: str) -> List:
+        """
+        Extract file paths from url
+
+        Args:
+        url (str): website
+
+        Returns:
+        file_paths (list): list of '.csv.gz' file paths
+        """
+
+        response = requests.get(url)
+        # Will raise an HTTPError
+        # if the HTTP request was unsuccessful
+        response.raise_for_status()
+        data = response.json()
+        # extract '.csv.gz' file paths
+        file_paths = [item["path"] for item in data["payload"]["tree"]["items"]]
+
+        return file_paths
+
+    @staticmethod
+    def filter_dates(file_list: List, time_period: Tuple[str, str]) -> List[str]:
+        """
+        Select dates within the interval defined by `time_period`
+
+        Args:
+
+            file_list (List): list of file paths
+            time_period (Tuple): time interval of the time t0 -> t1
+
+        Returns:
+            filtered_paths (List): list of filtered paths
+
+        Raises:
+            ValueError: if the time_period is not within the
+            range of dates in the file
+        """
+        start = datetime.strptime(time_period[0], "%Y-%m-%d")
+        end = datetime.strptime(time_period[1], "%Y-%m-%d")
+
+        # regular expression to extract dates from string
+        pattern = re.compile(r"(\d{4}-\d{2}-\d{2})")
+
+        all_dates = []
```

Comment:
Just to clarify the contents of this list, I would add a type hint here.

```suggestion
        all_dates: list[datetime.datetime] = []
```